src/applications/personalization/profile360/util/index.js

```diff
@@ -34,15 +34,15 @@ export async function getData(apiRoute, options) {
 const hasErrorMessage = (errors, errorKey, errorText) => {
   if (errorText) {
     return errors.some(err =>
-      err.meta.messages.some(
+      err?.meta?.messages?.some(
         message =>
           message.key === errorKey &&
-          message.text.toLowerCase().includes(errorText.toLowerCase()),
+          message.text?.toLowerCase().includes(errorText.toLowerCase()),
       ),
     );
   }
   return errors.some(err =>
-    err.meta.messages.some(message => message.key === errorKey),
+    err?.meta?.messages?.some(message => message.key === errorKey),
```

Comment:
Optional chaining is still confusing to me when it goes this deep. If `err` is empty, does this produce `hasErrorMessage = true`? I

Reply:
You're too fast. I updated the code. But `err?.meta` means that is will check the value of `err.meta` if `err` is a valid object. So `hasErrorMessage` would be `false` since `errors.some()` would not evaluate to `true`.